blueprints/third-party-solutions/gitlab-runner/main.tf

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  role_id            = "projects/${module.project.project_id}/roles/${local.role_name}"
+  role_name          = "gitlab_runner_manager_role"
+  runner_config_type = [for key, value in var.gitlab_runner_config.executors_config : key if value != null][0]
+  runner_startup_script_config = {
+    gitlab_hostname      = var.gitlab_config.hostname
+    gitlab_ca_cert       = base64encode(var.gitlab_config.ca_cert_pem)
+    token                = var.gitlab_runner_config.authentication_token
+    gitlab_runner_config = base64encode(templatefile("${path.module}/assets/config/${local.runner_config_type}_config.toml.tpl", var.gitlab_runner_config.executors_config[local.runner_config_type]))
+    gitlab_executor_type = replace(local.runner_config_type, "_", "-")
+  }
+}
+
+resource "google_service_account_iam_member" "admin-account-iam" {
```

Comment:
I'd create this SA using the service account module and use one of the IAM options to create this binding.

Reply:
Done thanks!